apps/vc-api/src/did/did.service.ts

```diff
@@ -46,7 +46,19 @@ export class DIDService {
 
   public async generateKeyDID(): Promise<DIDDocument> {
     const keyDescription = await this.keyService.generateKey({ type: keyType.ed25519 });
-    const key = await this.keyService.getPublicKeyFromKeyId(keyDescription.keyId);
+    return await this.registerKeyDID(keyDescription.keyId);
+  }
+
+  /**
+   * Register a did:key for a key which is known to the server.
```

Comment:
My idea is that from the DID module's perspective, the key just needs to be available from the Key module. So the key could be on the server, but it could also be in another location (e.g. in a cloud key vault). Good question though, I will update the comment to be more clear.